src/assets.zig

```diff
@@ -161,7 +149,38 @@ pub fn readAllZonFilesInAddons(
 		}
 	}
 }
+fn createAssetStringID(
+	externalAllocator: NeverFailingAllocator,
+	addonName: []const u8,
+	fileBaseName: []const u8,
+	relativeFilePath: []const u8,
```

Comment:
If the `relativeFilePath` ends with the file name, then we probably don't even need to use `fileBaseName`. That would simplify the code below a bit.

Reply:
We don't know what path separator is used, so while technically we could do that, it would complicate code and platform dependent.

To make the matter more complicated, based on this: https://en.m.wikipedia.org/wiki/Filename#Comparison_of_filename_limitations

Apparently \ is allowed in file names on unix systems, so you can't split on it if you are on Linux. And now we are reimplementing system dependent path handling.

Reply:
What does this have to do with the path separator? You are only looking at the last dot in the entire thing anyways.
And if there is no . in the file name then it can't even get to the function call.

Reply:
Rigth. I missed the fact that we no longer care about the other dots.